File: app/src/main/java/dahoum/wales/access_app/models/Visit.java
package dahoum.wales.access_app.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Visit {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("type")
    @Expose
    private String type;
    @SerializedName("startTime")
    @Expose
    private String startTime;
    @SerializedName("endTime")
    @Expose
    private String endTime;
    @SerializedName("visitors")
    @Expose
    private Integer visitors;
    @SerializedName("occupiedSlots")
    @Expose
    private Integer occupiedSlots;
    @SerializedName("maxSlots")
    @Expose
    private Integer maxSlots;
    @SerializedName("slotId")
    @Expose
    private String slotId;
    @SerializedName("placeId")
    @Expose
    private String placeId;
    private int viewType;

    public Visit(String startTime, int viewType) {
        this.startTime = startTime;
        this.viewType = viewType;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getType() {
        return type;
    }

    public String getStartTime() {
        return startTime;
    }

    public String getEndTime() {
        return endTime;
    }

    public Integer getOccupiedSlots() {
        return occupiedSlots;
    }

    public Integer getMaxSlots() {
        return maxSlots;
    }

    public int getViewType() {
        return viewType;
    }

    public Integer getVisitors() {
        return visitors;
    }

    public String getSlotId() {
        return slotId;
    }

    public String getPlaceId() {
        return placeId;
    }
}

File: resources/src/main/java/app/downloadaccess/resources/models/Visit.kt
package app.downloadaccess.resources.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Visit(
    @field:Expose @field:SerializedName("startTime") val startTime: String,
    val viewType: Int
) {
    @SerializedName("name")
    @Expose
    var name: String? = null

    @SerializedName("type")
    @Expose
    val type: String? = null

    @SerializedName("endTime")
    @Expose
    val endTime: String? = null

    @SerializedName("visitors")
    @Expose
    val visitors: Int? = null

    @SerializedName("occupiedSlots")
    @Expose
    val occupiedSlots: Int? = null

    @SerializedName("maxSlots")
    @Expose
    val maxSlots: Int? = null

    @SerializedName("slotId")
    @Expose
    val slotId: String? = null

    @SerializedName("placeId")
    @Expose
    val placeId: String? = null

}